Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -176,35 +177,80 @@ class Frontend {
+      // default configuration.
+      return Configuration()
     }
+    
+    // Now that we have the configURL, we're going to load it and validate the rules in it.
+    do {
+      // Ok to force unwrap because the only branch where configURL would not be set returned early.
+      let configuration = try configurationLoader.configuration(at: configURL!)
+      try self.verifyConfigurationRules(for: configuration, at: configURL!)
 
-    // If no explicit configuration file path was given but a `.swift` source file path was given,
-    // then try to load the configuration by inferring it based on the source file path.
-    if let swiftFileURL = swiftFileURL {
-      do {
-        if let configuration = try configurationLoader.configuration(forSwiftFileAt: swiftFileURL) {
-          return configuration
-        }
-        // Fall through to the default return at the end of the function.
-      } catch {
+      return configuration
+    } catch {
+      if let swiftFileURL {
         diagnosticsEngine.emitError(
           "Unable to read configuration for \(swiftFileURL.path): \(error.localizedDescription)")
-        return nil
+      } else {
+        diagnosticsEngine.emitError("Unable to read configuration: \(error.localizedDescription)")
       }
+      return nil
+    }
+  }
+
+  /// Checks if all the rules in the given configuration are supported by the registry.
+  /// If there are any rules that are not recognized by this version of swift-format, they are emitted as a warning.
+  ///
+  /// - parameters:
+  ///   - configuration: `Configuration` to verify
+  ///   - configURL: Configuration file `URL` to use in `Diagnostic.Location`
+  private func verifyConfigurationRules(for configuration: Configuration, at configURL: URL) throws {
+    // If any rules in the decoded configuration are not supported by the registry,
+    // emit them into the diagnosticsEngine as warnings.
+    let invalidRules = configuration.rules.filter { !RuleRegistry.rules.keys.contains($0.key) }
+
+    let configData = try Data(contentsOf: configURL)
+    guard let configString = String(data: configData, encoding: .utf8) else {
+      diagnosticsEngine.emitError("Unable to read the configuration file at \(configURL.description)")
+      return
+    }
+
+    for rule  in invalidRules {
+      let location = diagnosticLocationFor(substring: rule.key, in: configString, filePath: configURL.relativePath)
+      diagnosticsEngine.emitWarning(
+        "Configuration contains an unrecognized rule \(rule.key)",
+        location: location
+      )
+    }
+  }
+
+  /// Returns a `Diagnostic.Location` of the substring in an array of Strings.
+  /// It's intented to find the location of a configuration Rule in an array of Strings representing the encoded JSON swift-format configuration file.
+  private func diagnosticLocationFor(substring: String, in configuration: String, filePath: String) -> Diagnostic.Location? {
```

Comment:
For what it's worth, I don't think this is really necessary. If someone mistypes a rule name, it won't take them long to figure out where in the configuration it is. And if we could get the information during decoding, it _might_ be useful (still questionable though), but since this is just doing raw string searches, it's too prone to false positives if the invalid rule name matches something somewhere else in the configuration string.

Can you remove this and just emit a warning with no location?